src/main/java/bgu/spl/mics/MessageBrokerImpl.java

```diff
@@ -38,6 +38,7 @@ private MessageBrokerImpl() {
 	 @post: m's queue will receive Event<T> objects.
 	 */
 	public <T> void subscribeEvent(Class<? extends Event<T>> type, Subscriber m) {
+		if(_eventSubscribers.get(type) == null) _eventSubscribers.put(type, new ConcurrentLinkedQueue<>());
```

Comment:
will never happen
one should not subscribe to event before it was registered. the bug is not here but in an other place

Reply:
my bad again. you create a queue for a broadcast and not a subscriber. good job.
but, add a comment about it being thread-safe